src/App.tsx

```diff
@@ -1,17 +1,137 @@
-import React from 'react';
+import React, { useState } from 'react';
+import ToDoList from './ToDoList';
 import './App.css';
 
 import users from './api/users';
+import todos from './api/todos';
 
 const App: React.FC = () => {
+  const [selectedUser, selectUser] = useState('no-user');
+  const [currentTitle, setTitle] = useState('');
+  const [toDoStatus, setToDoStatus] = useState(false);
+  const [todosCopy, handleTodos] = useState([...todos]);
+  const [lastTodoId, increaseId] = useState(todos[todos.length - 1].id + 1);
+  const [isError, setError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleChange = (event: any) => {
+    const { value, id } = event.target;
+
+    setError(false);
+
+    switch (id) {
+      case 'usersSelect': {
+        selectUser(value);
+        break;
+      }
+
+      case 'todoTitle': {
+        setTitle(value);
+        break;
+      }
+
+      case 'toDoStatus': {
+        setToDoStatus(!toDoStatus);
+        break;
+      }
+
+      default: {
+        break;
+      }
+    }
+  };
+
+  const resetForm = () => {
+    selectUser('no-user');
+    setTitle('');
+    setToDoStatus(false);
+  };
+
+  const addToDoItem = (event: React.FormEvent) => {
+    event.preventDefault();
+    const currentUser = users.find(user => user.name === selectedUser);
+
+    if (selectedUser === 'no-user') {
+      setErrorMessage('Please choose a user');
+      setError(true);
+
+      return null;
+    }
+
+    if (currentTitle === '') {
+      setErrorMessage('Please enter the title');
+      setError(true);
+
+      return null;
+    }
+
+    if (currentUser === undefined) {
+      resetForm();
+
+      return null;
+    }
+
+    handleTodos(
```

Comment:
```suggestion
    handleTodos((prev) => 
```
Используй setState с коллбеком

Reply:
как я могу использовать setState, если это функциональный компонент?

Reply:
`handleTodos` это функция, которая возвращается из `useState` хука, она и есть setState в функциональном компоненте